codemods/run-codemod.ts

```diff
@@ -0,0 +1,191 @@
+    const options = [
+        { name: "src (./src)", value: predefinedPaths.Src },
+        { name: "current directory (./)", value: predefinedPaths.Codebase },
+        { name: "custom", value: "custom" },
+    ];
+
+    const selectedOption = await select({
+        message: "Choose an output path:",
+        choices: options,
+    });
+
+    if (selectedOption === "custom") {
+        const customPath = await input({
+            message: "Enter your custom path:",
+        });
+
+        if (!customPath) {
+            console.log("No custom path provided. Canceling.");
+            return null;
+        }
+
+        const resolvedPath = path.resolve(customPath);
+
+        if (!fs.existsSync(resolvedPath)) {
+            console.error(
+                `The path "${resolvedPath}" does not exist. Please try again.`
+            );
+            return null;
+        }
+
+        return resolvedPath;
+    }
+
+    return selectedOption;
+}
+
+async function chooseTheme(): Promise<Theme | null> {
+    const themeOptions = [
+        { name: "LifeSG", value: Theme.LifeSG },
+        { name: "BookingSG", value: Theme.BookingSG },
+    ];
+
+    const selectedTheme = await select({
+        message: "Select the theme to follow for Color migration:",
+        choices: themeOptions,
+    });
+
+    return selectedTheme;
+}
+
+// Choose and run the codemods
+async function main(): Promise<void> {
+    const codemods = listCodemods();
+    if (codemods.length === 0) {
+        console.log("No codemod scripts found.");
+        return;
+    }
+
+    try {
+        const selectedCodemods = await checkbox({
+            required: true,
+            message: "Select codemods to run:",
+            choices: codemods,
+        });
+
+        let selectedTheme: Theme | null = null;
+        if (selectedCodemods.includes(Codemod.MigrateColour)) {
+            selectedTheme = await chooseTheme();
+        }
+
+        const outputPath = await chooseOutputPath();
+        if (!outputPath) {
+            console.log("No output path selected or provided.");
+            return;
+        }
+
+        const finalConfirmationMessage = `You are about to run the following codemods: ${selectedCodemods.join(
+            ", "
+        )}\nOutput path: ${outputPath}${
+            selectedTheme
+                ? `\nSelected theme for "migrate-colour": ${selectedTheme}`
+                : ""
+        }.\nDo you want to proceed?`;
+
+        const finalConfirmation = await confirm({
+            message: finalConfirmationMessage,
+            default: true,
+        });
+
+        if (!finalConfirmation) {
+            return;
+        }
+
+        runCodemods(
+            selectedCodemods,
+            outputPath,
+            selectedTheme ? { "migrate-colour": selectedTheme } : undefined
+        );
+    } catch (error) {
+        if (error.name === "ExitPromptError") {
+            console.log("Selection process interrupted.");
+        }
+    }
```

Comment:
for unhandled errors, make sure to throw them upwards to avoid unintentionally swallowing them

```
throw error;
```